Mention and Main Component

powerseed opened the pull request:
Developed Mention and Main Component with dummy content.

powerseed committed:
Mention and Main Component

Developed Mention and Main Component with dummy content.

grogsy reviewed (approved):
checked out the UI, looks great!

aeciorc commented on client/package.json:
```diff
@@ -3,11 +3,14 @@
-    "@material-ui/core": "^4.1.3",
+    "@material-ui/core": "^4.10.2",
+    "@material-ui/icons": "^4.9.1",
+    "fontsource-roboto": "^2.1.4",
```
The font roboto is already installed. Checkout index.html:
```html
   <link
      rel="stylesheet"
      type="text/css"
      href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Material+Icons"
    />
```

aeciorc commented on client/src/App.js:
```diff
@@ -1,17 +1,19 @@
     <MuiThemeProvider theme={theme}>
       <BrowserRouter>
-        <Route path="/" component={LandingPage} />
+        <Route path="/landing" component={LandingPage} />
```
You can delete this component since we don't have a landing page

aeciorc commented on client/src/Components/Header.js:
```diff
@@ -0,0 +1,83 @@
+
+const useStyles = makeStyles((theme) => ({
+  typography: {
+    fontFamily: theme.typography.fontFamily,
```
You don't need this, actually. Because it's set on the theme.js, every typography will already have the fontFamily set there

aeciorc commented on client/src/Components/Header.js:
```diff
@@ -0,0 +1,83 @@
+    fontFamily: theme.typography.fontFamily,
+  },
+  palette: {
+    background: theme.palette.primary.main,
```
Same for this. It will already use the palette for the theme primary color by default

aeciorc commented on client/src/Components/Header.js:
```diff
@@ -0,0 +1,83 @@
+          <Grid container spacing={0}>
+            <Grid item xs={4} style={{ display: "flex", alignItems: "center" }}>
+              <Typography className={classes.typography} variant="h6">
+                <span style={{ color: "white" }}>mentions</span>
+                <span style={{ color: "#191970" }}>crawler.</span>
```
Since we have a style object, I would just use className everywhere in this component, rather than inline.

aeciorc commented on client/src/Components/Header.js:
```diff
@@ -0,0 +1,83 @@
+      </AppBar>
+    </div>
+  );
+}
```
The navbar looks great 👍

aeciorc commented on client/src/Components/Mention.js:
```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import Card from "@material-ui/core/Card";
+import CardActionArea from "@material-ui/core/CardActionArea";
+import CardContent from "@material-ui/core/CardContent";
+import CardMedia from "@material-ui/core/CardMedia";
+import Typography from "@material-ui/core/Typography";
+import Grid from "@material-ui/core/Grid";
```
You can group these imports:
```js
import {
  Card,
  CardActionArea,
  CardContent,
  CardMedia,
  Typography,
  Grid,
} from "@material-ui/core";

```

aeciorc commented on client/src/pages/Main.js:
```diff
@@ -0,0 +1,121 @@
+import { withStyles } from "@material-ui/core/styles";
+
+const styles = (theme) => ({
+  typography: {
+    fontFamily: theme.typography.fontFamily,
+  },
+  palette: {
+    background: theme.palette.primary.main,
+  },
+});
```
Same comment here as in Header.js

aeciorc commented on client/src/pages/Main.js:
```diff
@@ -0,0 +1,121 @@
+            >
+              <h1 className={classes.typography}>My mentions</h1>
+              <div style={{ height: "2.5em", width: "30%" }}>
+                <SwitchSelector
+                  // onChange={onChange}
+                  options={[
+                    {
+                      label: "Most recent",
+                      value: {
+                        foo: true,
+                      },
+                      selectedBackgroundColor: "#4169E1",
+                      selectedFontColor: "white",
+                    },
+                    {
+                      label: "Most popular",
+                      value: "bar",
+                      selectedBackgroundColor: "#4169E1",
+                      selectedFontColor: "white",
+                    },
+                  ]}
+                  backgroundColor={"#D6EAF8"}
+                  fontColor={"#4169E1"}
+                />
```
Try to make this a little flatter, and give the font more weight

aeciorc commented on client/src/pages/Main.js:
```diff
@@ -0,0 +1,121 @@
+class Main extends Component {
+  render() {
+    const { classes } = this.props;
+
```
Notice that the background in this page is actually a light shade of blue. You can use a color picker to get the exact tone rgb

aeciorc commented on client/src/Components/Mention.js:
```diff
@@ -0,0 +1,64 @@
+
+export default function Mention(props) {
+  return (
+    <Card style={{ height: "200px" }}>
```
Small improvement: Notice that the card has a shadow by default, but the one on the mockup doesn't

powerseed committed:
Fix issues

1. Removed landing.js
2. Grouped imports
3. Added background color for main page.
4. Used style object instead inline style.

powerseed commented:
@aeciorc Fixed the issues you pointed out. Please review it. Thanks
